<?xml version="1.0" ?>
<launch>
  <arg name="use_rviz" default="true" />
  <arg name="rviz_config" default="$(find bimanual_planning_ros)/config/rviz_kobo_sim_experiments.rviz" />
  <arg name="sim" default="true" />
  <arg name="bag" default="false" />
  <arg name="panda_left_id" value="panda_left" />
  <arg name="panda_right_id" value="panda_right" />

  <node name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" args="-d $(arg rviz_config)" output="screen" if="$(arg use_rviz)" />

  <include file="$(find bimanual_planning_ros)/launch/planning_node.launch"/>
  
  <param name="/panda_dual/dual_panda_costp_controller/bimanual_planning/robot_type" value="vrep" if="$(arg sim)"/>
  
  <group ns="panda_dual">
    <param name="robot_description" command="xacro --inorder $(find blaise_description)/robots/blaise.urdf.xacro simulation:=false with_hands:=true"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
    <node name="joint_state_publisher" type="joint_state_publisher" pkg="joint_state_publisher" output="screen">
      <rosparam param="source_list" subst_value="True">[dual_panda_costp_controller/$(arg panda_left_id)/joint_states,
      dual_panda_costp_controller/$(arg panda_right_id)/joint_states]
      </rosparam>
      <param name="rate" value="30"/>
    </node>
    <group ns="dual_panda_costp_controller">
      <node pkg="bimanual_planning_ros" type="dynamic_obstacle_node" name="dynamic_obstacle_node" output="screen" />
    </group>
  </group>

  <node pkg="rosbag" type="record" name="rosbag_record" args="record /panda_dual/dual_panda_costp_controller/visualization_marker_array /tf /tf_static /panda_dual/dual_panda_costp_controller/predicted_paths /panda_dual/dual_panda_costp_controller/commanded_path" if="$(arg bag)" />

</launch>
